bin/alluxio

```diff
@@ -41,29 +41,74 @@ COMMAND=$1
-    if [[ `uname -a` == Darwin* ]]; then
+    if [[ $(uname -s) == Darwin ]]; then
       # osx sed wants an argument to -i, use .bootstrap.bk
       ALLUXIO_SED="sed -i .bootstrap.bk"
-      TOTAL_MEM=`sysctl hw.memsize | cut -d ' ' -f2`
+      TOTAL_MEM=$(sysctl hw.memsize | cut -d ' ' -f2)
       TOTAL_MEM=$[TOTAL_MEM / 1024]
       TOTAL_MEM=$[TOTAL_MEM * 2 / 3]
     else
       ALLUXIO_SED="sed -i"
-      TOTAL_MEM=`awk '/MemTotal/{print $2}' /proc/meminfo`
+      TOTAL_MEM=$(awk '/MemTotal/{print $2}' /proc/meminfo)
       TOTAL_MEM=$[TOTAL_MEM * 2 / 3]
     fi
 
+    local dst="${ALLUXIO_CONF_DIR}/alluxio-env.sh"
     # Create a default config that can be overridden later
-    cp ${ALLUXIO_CONF_DIR}/alluxio-env.sh.template ${ALLUXIO_CONF_DIR}/alluxio-env.sh
-    ${ALLUXIO_SED} "s/ALLUXIO_MASTER_ADDRESS=localhost/ALLUXIO_MASTER_ADDRESS=$1/" ${ALLUXIO_CONF_DIR}/alluxio-env.sh
-    ${ALLUXIO_SED} "s/ALLUXIO_WORKER_MEMORY_SIZE=1GB/ALLUXIO_WORKER_MEMORY_SIZE=${TOTAL_MEM}KB/" ${ALLUXIO_CONF_DIR}/alluxio-env.sh
+    cp "${ALLUXIO_CONF_DIR}/alluxio-env.sh.template" ${dst}
+
+    # Set the master hostname
+    ${ALLUXIO_SED} "s/XXX_MASTER_ADDRESS_XXX/$1/" ${dst}
+
+    # Set the worker memory size
+    ${ALLUXIO_SED} "s/XXX_MEMORY_SIZE_XXX/${TOTAL_MEM}KB/" ${dst}
+
+    # Set under storage specific configuration
+    case ${ufs} in
+      local)
+        ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/\$\{ALLUXIO_HOME\}\/under-storage/" ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+        ;;
+      hdfs)
+        ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/hdfs:\/\/localhost:9000\/alluxio/" ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+        ;;
+      s3)
+        ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/s3n:\/\/bucket\/directory/" ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+        ;;
+      gcs)
+        ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/gs:\/\/bucket\/directory/" ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+        ;;
+      glusterfs)
+        ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/glusterfs:\/\/\//" ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a\\
+        export ALLUXIO_UNDERFS_GLUSTER_VOLUMES=\$\{ALLUXIO_UNDERFS_GLUSTER_VOLUMES:-alluxio_vol\}\\
+        " ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
+        export ALLUXIO_UNDERFS_GLUSTER_MOUNTS=\${ALLUXIO_UNDERFS_GLUSTER_MOUNTS:-\/vol\}\\
+        " ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/a \\
+        export ALLUXIO_UNDERFS_GLUSTER_MR_DIR=\${ALLUXIO_UNDERFS_GLUSTER_MR_DIR:-\glusterfs:\/\/\/mapred\/system\}\\
+        " ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+        ;;
+      swift)
+        ${ALLUXIO_SED} "s/XXX_UNDER_STORAGE_XXX/swift:\/\/swift-container/" ${dst}
+        ${ALLUXIO_SED} "/XXX_ADDITIONAL_SETTINGS_XXX/d" ${dst}
+        ;;
+    esac
```

Comment:
Could you be more specific? Currently, we have a number of template files which contain duplicated (and in some cases out of date) code and except for one,are not actually used by the bootstrap command.